libminifi/include/utils/GeneralUtils.h

```diff
@@ -34,6 +34,16 @@ constexpr T intdiv_ceil(T numerator, T denominator) {
       : numerator / denominator + (numerator % denominator != 0));
 }
 
+/**
+ * safely converts unique_ptr from one type to another
+ * if conversion succeeds, an desired valid unique_ptr is returned, the "from" object released and invalidated
+ * if conversion fails, an empty unique_ptr is returned, the "from" object staying untouched
+ */
+template <typename T_To, typename T_From>
+std::unique_ptr<T_To> dynamic_unique_cast(std::unique_ptr<T_From>& obj) {
```

Comment:
I'm a bit concerned about keeping the original deleter of the casted object. Maybe we should use this proposed version: https://stackoverflow.com/questions/21174593/downcasting-unique-ptrbase-to-unique-ptrderived?rq=1

Reply:
With this signature, it would simply not work for unique_ptrs with custom deleters. With the standard deleter, it doesn't matter.

Reply:
Good point, although maybe we could still include it just to have a more generic implementation.

Reply:
I think we should only include it if we use it.

Reply:
I don't insist on it, we can keep it as it for now

Reply:
Okay, with obj argument now being &&.